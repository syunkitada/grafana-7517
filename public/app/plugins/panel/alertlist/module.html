<div class="panel-alert-list" style="{{ctrl.contentHeight}}">
  <section class="card-section card-list-layout-list" ng-if="ctrl.panel.show === 'current'">
    <ol class="card-list">
      <li class="card-item-wrapper" ng-repeat="alert in ctrl.currentAlerts">
        <div class="card-item card-item--alert">
          <div class="card-item-body">
            <div class="card-item-details">
              <div class="card-item-notice">
                <a href="dashboard/{{alert.dashboardUri}}?panelId={{alert.panelId}}&fullscreen&edit&tab=alert">
                  {{alert.name}}
                </a>
              </div>
              <div class="card-item-sub-name">
                <span class="alert-list-item-state {{alert.stateModel.stateClass}}">
                  <i class="{{alert.stateModel.iconClass}}"></i>
                  {{alert.stateModel.text}}
                </span> for {{alert.newStateDateAgo}}
              </div>
              <div class="small muted" ng-show="alert.executionError">
                {{alert.executionError}}
              </div>
            </div>
          </div>
        </div>
      </li>
    </ol>
  </section>

  <section class="card-section card-list-layout-list" ng-if="ctrl.panel.show === 'changes'">
    <ol class ="card-list">
      <li class="card-item-wrapper" ng-repeat="al in ctrl.alertHistory">
        <div class="alert-list card-item card-item--alert">
          <div class="alert-list-body">
            <div class="alert-list-icon alert-list-item-state {{al.stateModel.stateClass}}">
              <i class="{{al.stateModel.iconClass}}"></i>
            </div>
            <div class="alert-list-main">
              <p class="alert-list-title">{{al.title}}</p>
              <p class="alert-list-state alert-list-text {{al.stateModel.stateClass}}">{{al.stateModel.text}}</p>
              <p class="alert-list-text">{{al.info}}</p>
            </div>
          </div>
          <div class="alert-list-footer">
            <p class="alert-list-text">{{al.time}}</p>
            <p class="alert-list-text">Img Link -></p>
          </div>
        </div>
      </li>
    </ol>
  </section>
</div>
