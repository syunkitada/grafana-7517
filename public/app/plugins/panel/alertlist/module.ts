///<reference path="../../../headers/common.d.ts" />

import _ from 'lodash';
import moment from 'moment';
import alertDef from '../../../features/alerting/alert_def';
import config from 'app/core/config';
import {PanelCtrl} from 'app/plugins/sdk';

class AlertListPanel extends PanelCtrl {
  static templateUrl = 'module.html';

  showOptions = [
    {text: 'Current state', value: 'current'},
    {text: 'State changes', value: 'changes'},
  ];

  currentAlerts: any = [];
  alertHistory: any = [];
  // Set and populate defaults
  panelDefaults = {
    show: 'current'
  };

  /** @ngInject */
  constructor($scope, $injector, private $location, private backendSrv) {
    super($scope, $injector);
    _.defaults(this.panel, this.panelDefaults);

    this.events.on('init-edit-mode', this.onInitEditMode.bind(this));
    this.events.on('render',  this.onRender.bind(this));
    this.events.on('refresh', this.onRender.bind(this));
  }

  onRender() {
    if (this.panel.show === 'current') {
      this.getCurrentAlertState();
    }

    if (this.panel.show === 'changes') {
      this.getAlertHistory();
    }
  }

  getAlertHistory() {
    this.backendSrv.get(`/api/alert-history?dashboardId=32&panelId=1`)
      .then(res => {
        this.alertHistory = _.map(res, al => {
          al.time = moment(al.timestamp).format('MMM D, YYYY HH:mm:ss');
          al.stateModel = alertDef.getStateDisplayModel(al.newState);
          return al;
        });
      });
  }

  getCurrentAlertState() {
    this.backendSrv.get(`/api/alerts`)
      .then(res => {
        this.currentAlerts = _.map(res, al => {
          al.stateModel = alertDef.getStateDisplayModel(al.state);
          al.newStateDateAgo = moment(al.newStateDate).fromNow().replace(" ago", "");

          return al;
        });
      });
  }

  onInitEditMode() {
    this.addEditorTab('Options', 'public/app/plugins/panel/alertlist/editor.html');
  }
}

export {
  AlertListPanel,
  AlertListPanel as PanelCtrl
}
