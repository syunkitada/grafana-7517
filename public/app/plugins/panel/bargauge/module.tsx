import { sharedSingleStatPanelChangedHandler } from '@grafana/ui';
import { PanelPlugin } from '@grafana/data';
import { BarGaugePanel } from './BarGaugePanel';
import { BarGaugeOptions, defaults } from './types';
import { addStandardDataReduceOptions } from '../stat/types';
import { BarGaugePanelEditor } from './BarGaugePanelEditor';
import { barGaugePanelMigrationHandler } from './BarGaugeMigrations';

export const plugin = new PanelPlugin<BarGaugeOptions>(BarGaugePanel)
  .setDefaults(defaults)
  .setEditor(BarGaugePanelEditor)
  .setPanelOptions(builder => {
    addStandardDataReduceOptions(builder);

    builder
      .addRadio({
        id: 'displayMode',
        name: 'Display mode',
        description: 'Controls the bar style',
        settings: {
          options: [
            { value: 'basic', label: 'Basic' },
            { value: 'gradient', label: 'Gradient' },
            { value: 'lcd', label: 'Retro LCD' },
          ],
        },
      })
      .addBooleanSwitch({
        id: 'showUnfilled',
        name: 'Show unfilled area',
        description: 'When enabled renders the unfilled region as gray',
      });
  })
  .setPanelChangeHandler(sharedSingleStatPanelChangedHandler)
  .setMigrationHandler(barGaugePanelMigrationHandler)
  .useStandardFieldConfig();
