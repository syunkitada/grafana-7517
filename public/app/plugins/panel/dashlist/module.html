<div class="dashlist">
  <div class="dashlist-section">
    <h6 class="dashlist-section-header">
      Getting Started with Grafana
      <button class="dashlist-CTA-close-btn"><i class="fa fa-remove"></i></button>
    </h6>
    <div class="dashlist-flex-container">
      <div class="dashlist-panel-CTA-flex-container--left">
        <h5 class="dashlist-next-up">Next up:</h5>
        <span class="dashlist-title">Create your first data source</span>
        <p class="dashlist-next-up-blurb">If Grafana dashboards are like the eyes, man, datasources are the heart. Cosmic.</p>
        <button class="btn btn-success">Add Data Source</button>
      </div>
      <ul class="dashlist-panel-CTA-flex-container--right">
        <li class="dashlist-panel-CTA-items">
          <div class="dashlist-icons-container">
              <i class="icon-gf icon-gf-check dashlist-icon-success"></i>
          </div>
          <span class="dashlist-blurb dashlist-blurb-success">Install Grafana like a boss.</span>
        </li>
        <li class="dashlist-panel-CTA-items dashlist-icons-container--active">
          <div class="dashlist-icons-container">
              <i class="icon-gf icon-gf-datasources dashlist-icon-active"></i>
          </div>
          <a href="#" class="dashlist-blurb">Add your first data source.</a>
        </li>
        <li class="dashlist-panel-CTA-items">
          <div class="dashlist-icons-container">
              <i class="icon-gf icon-gf-dashboard dashlist-icon-upcoming"></i>
          </div>
          <a href="#" class="dashlist-blurb dashlist-blurb-upcoming">Create your first dashboard.</a>
        </li>
        <li class="dashlist-panel-CTA-items">
          <div class="dashlist-icons-container">
              <i class="icon-gf icon-gf-users dashlist-icon-upcoming"></i>
          </div>
          <a href="#" class="dashlist-blurb dashlist-blurb-upcoming">Invite your team.</a>
        </li>
        <li class="dashlist-panel-CTA-items">
          <div class="dashlist-icons-container">
              <i class="icon-gf icon-gf-grafana-icon dashlist-icon-upcoming"></i>
          </div>
          <a href="#" class="dashlist-blurb dashlist-blurb-upcoming">Extend on Grafana.net.</a>
        </li>
      </ul>
    </div>
  </div>
</div>
<div class="dashlist" ng-repeat="group in ctrl.groups">
  <div class="dashlist-section" ng-if="group.show">
    <h6 class="dashlist-section-header" ng-show="ctrl.panel.headings">
      {{group.header}}
    </h6>
    <div class="dashlist-item" ng-repeat="dash in group.list">
      <a class="dashlist-link dashlist-link-{{dash.type}}" href="dashboard/{{dash.uri}}">
        <span class="dashlist-title">
          {{dash.title}}
        </span>
        <span class="dashlist-star">
          <i class="fa" ng-class="{'fa-star': dash.isStarred, 'fa-star-o': dash.isStarred === false}"></i>
        </span>
      </a>
    </div>
  </div>
</div>
