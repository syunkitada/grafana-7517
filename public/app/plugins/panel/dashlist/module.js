///<reference path="../../../headers/common.d.ts" />
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var lodash_1 = require('lodash');
var sdk_1 = require('app/plugins/sdk');
var impression_store_1 = require('app/features/dashboard/impression_store');
// Set and populate defaults
var panelDefaults = {
    mode: 'starred',
    query: '',
    limit: 10,
    tags: [],
    recent: false,
    search: false,
    starred: true
};
var DashListCtrl = (function (_super) {
    __extends(DashListCtrl, _super);
    /** @ngInject */
    function DashListCtrl($scope, $injector, backendSrv) {
        _super.call(this, $scope, $injector);
        this.backendSrv = backendSrv;
        lodash_1["default"].defaults(this.panel, panelDefaults);
        if (this.panel.tag) {
            this.panel.tags = [this.panel.tag];
            delete this.panel.tag;
        }
        this.events.on('refresh', this.onRefresh.bind(this));
        this.events.on('init-edit-mode', this.onInitEditMode.bind(this));
    }
    DashListCtrl.prototype.onInitEditMode = function () {
        this.editorTabIndex = 1;
        this.modes = ['starred', 'search', 'recently viewed'];
        this.addEditorTab('Options', 'public/app/plugins/panel/dashlist/editor.html');
    };
    DashListCtrl.prototype.onRefresh = function () {
        var promises = [];
        if (this.panel.recent) {
            promises.push(this.getRecentDashboards());
        }
        if (this.panel.starred) {
            promises.push(this.getStarred());
        }
        if (this.panel.search) {
            promises.push(this.getSearch());
        }
        return Promise.all(promises)
            .then(this.renderingCompleted.bind(this));
    };
    DashListCtrl.prototype.getSearch = function () {
        var _this = this;
        var params = {
            limit: this.panel.limit,
            query: this.panel.query,
            tag: this.panel.tags
        };
        return this.backendSrv.search(params).then(function (result) {
            _this.dashList = result;
            _this.renderingCompleted();
        });
    };
    DashListCtrl.prototype.getStarred = function () {
        var _this = this;
        var params = { limit: this.panel.limit, starred: "true" };
        return this.backendSrv.search(params).then(function (result) {
            _this.dashList = result;
            _this.renderingCompleted();
        });
    };
    DashListCtrl.prototype.getRecentDashboards = function () {
        var _this = this;
        var dashIds = lodash_1["default"].first(impression_store_1.impressions.getDashboardOpened(), this.panel.limit);
        return this.backendSrv.search({ dashboardIds: dashIds, limit: this.panel.limit }).then(function (result) {
            _this.dashList = dashIds.map(function (orderId) {
                return lodash_1["default"].find(result, function (dashboard) {
                    return dashboard.id === orderId;
                });
            }).filter(function (el) {
                return el !== undefined;
            });
        });
    };
    DashListCtrl.templateUrl = 'module.html';
    return DashListCtrl;
}(sdk_1.PanelCtrl));
exports.DashListCtrl = DashListCtrl;
exports.PanelCtrl = DashListCtrl;
