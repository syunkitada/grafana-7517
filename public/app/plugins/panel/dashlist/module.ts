///<reference path="../../../headers/common.d.ts" />

import _ from 'lodash';
import config from 'app/core/config';
import {PanelCtrl} from 'app/plugins/sdk';
import {impressions} from 'app/features/dashboard/impression_store';

 // Set and populate defaults
var panelDefaults = {
  mode: 'starred',
  query: '',
  limit: 10,
  tags: []
};

class DashListCtrl extends PanelCtrl {
  static templateUrl = 'module.html';

  dashList: any[];
  modes: any[];

  /** @ngInject */
  constructor($scope, $injector, private backendSrv) {
    super($scope, $injector);
    _.defaults(this.panel, panelDefaults);

    if (this.panel.tag) {
      this.panel.tags = [this.panel.tag];
      delete this.panel.tag;
    }

    this.events.on('refresh', this.onRefresh.bind(this));
    this.events.on('init-edit-mode', this.onInitEditMode.bind(this));
  }

  onInitEditMode() {
    this.editorTabIndex = 1;
    this.modes = ['starred', 'search', 'recently viewed'];
    this.addEditorTab('Options', 'public/app/plugins/panel/dashlist/editor.html');
  }

  onRefresh() {
    var params: any = {limit: this.panel.limit};

    if (this.panel.mode === 'recently viewed') {
      var dashIds = _.first(impressions.getDashboardOpened(), this.panel.limit);

      return this.backendSrv.search({dashboardIds: dashIds, limit: this.panel.limit}).then(result => {
        this.dashList = dashIds.map(orderId => {
          return _.find(result, dashboard => {
            return dashboard.id === orderId;
          });
        }).filter(el => {
          return el !== undefined;
        });

        this.renderingCompleted();
      });
    }

    if (this.panel.mode === 'starred') {
      params.starred = "true";
    } else {
      params.query = this.panel.query;
      params.tag = this.panel.tags;
    }

    return this.backendSrv.search(params).then(result => {
      this.dashList = result;
      this.renderingCompleted();
    });
  }
}

export {DashListCtrl, DashListCtrl as PanelCtrl}
