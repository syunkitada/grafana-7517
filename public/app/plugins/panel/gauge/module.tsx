import { defaultStandardFieldConfigProperties, PanelPlugin } from '@grafana/data';
import { GaugePanelEditor } from './GaugePanelEditor';
import { GaugePanel } from './GaugePanel';
import { GaugeOptions, defaults } from './types';
import { standardFieldConfigDefaults, addStandardDataReduceOptions } from '../stat/types';
import { gaugePanelMigrationHandler, gaugePanelChangedHandler } from './GaugeMigrations';

export const plugin = new PanelPlugin<GaugeOptions>(GaugePanel)
  .setDefaults(defaults)
  .setEditor(GaugePanelEditor)
  .setPanelOptions(builder => {
    addStandardDataReduceOptions(builder);

    builder
      .addBooleanSwitch({
        id: 'showThresholdLabels',
        name: 'Show threshold Labels',
        description: 'Render the threshold values around the gauge bar',
      })
      .addBooleanSwitch({
        id: 'showThresholdMarkers',
        name: 'Show threshold markers',
        description: 'Renders the thresholds as an outer bar',
      });
  })
  .setPanelChangeHandler(gaugePanelChangedHandler)
  .setMigrationHandler(gaugePanelMigrationHandler)
  .useStandardFieldConfig(defaultStandardFieldConfigProperties, standardFieldConfigDefaults);
