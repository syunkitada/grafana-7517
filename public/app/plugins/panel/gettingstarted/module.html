<div class="dashlist" ng-if="ctrl.checksDone">
  <div class="dashlist-section">
    <h6 class="dashlist-section-header">
      Getting Started with Grafana
      <button class="dashlist-cta-close-btn" ng-click="ctrl.dismiss()">
        <i class="fa fa-remove"></i>
      </button>
    </h6>
    <ul class="progress-tracker">
      <li class="progress-step completed">
        <span class="progress-marker"><i class="icon-gf icon-gf-check"></i></span>
        <span class="progress-text">Install Grafana</span>
      </li>
      <li class="progress-step" ng-class="ctrl.getStateClass(2)">
        <span class="progress-marker"><i class="icon-gf icon-gf-datasources"></i></span>
        <span class="progress-text">Create your first data source</span>
        <a class="btn progress-step-cta" href="datasources/new?gettingstarted">Add Data Source</a>
      </li>
      <li class="progress-step" ng-class="ctrl.getStateClass(3)">
        <span class="progress-marker"><i class="icon-gf icon-gf-dashboard"></i></span>
        <span class="progress-text">Create your first dashboard.</span>
        <a class="btn progress-step-cta" href="dashboard/new?gettingstarted">New Dashboard</a>
      </li>
      <li class="progress-step">
        <span class="progress-marker"><i class="icon-gf icon-gf-users"></i></span>
        <span class="progress-text">Invite your team.</span>
        <a class="btn progress-step-cta" href="org/users?gettingstarted">Invite Users</a>
      </li>
      <li class="progress-step">
        <span class="progress-marker"><i class="icon-gf icon-gf-apps"></i></span>
        <span class="progress-text">Install apps &amp; plugins</span>
        <a class="btn progress-step-cta" href="https://grafana.net/plugins?utm_source=grafana_getting_started">Explore Plugins</a>
      </li>
    </ul>
  </div>
</div>
