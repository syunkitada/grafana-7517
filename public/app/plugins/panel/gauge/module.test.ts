import { gaugePanelTypeChangedHook } from './module';

describe('Gauge Module', () => {
  describe('migrations', () => {
    it('should migrate from 6.0 settings to 6.1', () => {
      const v60 = {
        minValue: 50,
        maxValue: 60,
        showThresholdMarkers: true,
        showThresholdLabels: false,
        valueOptions: {
          prefix: 'a',
          suffix: 'z',
          decimals: 4,
          stat: 'avg',
          unit: 'ms',
        },
        valueMappings: [],
        thresholds: [{ index: 0, value: -Infinity, color: 'green' }, { index: 1, value: 80, color: 'red' }],
      };

      const after = gaugePanelTypeChangedHook(v60);
      expect((after.stat = 'avg'));
      expect(after).toMatchSnapshot();
    });
  });
});
