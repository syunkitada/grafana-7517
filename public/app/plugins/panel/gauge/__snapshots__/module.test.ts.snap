// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gauge Module migrations should migrate from 6.0 settings to 6.1 1`] = `
Object {
  "display": Object {
    "decimals": 4,
    "mappings": Array [],
    "prefix": "a",
    "stat": "avg",
    "suffix": "z",
    "thresholds": Array [
      Object {
        "color": "green",
        "index": 0,
        "value": -Infinity,
      },
      Object {
        "color": "red",
        "index": 1,
        "value": 80,
      },
    ],
    "unit": "ms",
  },
  "maxValue": 60,
  "minValue": 50,
  "showThresholdLabels": false,
  "showThresholdMarkers": true,
  "stat": "avg",
}
`;
